### PR TITLE
Stepping Speed Improvements

### PR DESCRIPTION
**Pull request by IhorNehrutsa**
1) Add STM32Cube generated SystemClock_Config_XXX() functions.
```
void SystemClock_Config_HSE_16M_SYSCLK_72M(void);
void SystemClock_Config_HSE_8M_SYSCLK_72M(void);
void SystemClock_Config_HSI_8M_SYSCLK_64M(void);
```
They are help to select needed SystemClock.

2) Add STM32Cube generated MCO_GPIO_Init() function.
```
void MCO_GPIO_Init(void);
```
I used MCO pin to detect HSE oscillator frequency. It is 8MHz.
https://github.com/CAP1Sup/Intellistep/issues/31 

3) Add #define CHECK_MCO_OUTPUT // use oscilloscope to measure frequency HSI, HSE, SYSCLK or PLLCLK/2

4) Add #define ENABLE_BLINK. Change blink frequency to 1Hz.

5) Copy 
legacy/Close_loop/src/User/main.c 
to VisualCode formatted 
legacy/Close_loop/src/User/main_formatted.c
It is for better readable.

### CODE REVIEW

**Comment by CAP1Sup** [comment]
Have you been able to test Serial with this? I remember that when using the HSE, the delay functions were way off, like 2x as fast as they should have been. When I tried to fix the frequency to be accurate, it would break the Serial communication.

**Comment by IhorNehrutsa** [comment]
No

File: src/user/main.cpp [hunk 1]
```diff
@@ -191,7 +211,12 @@ void setup() {
             #endif
 
             // We need a little delay to allow the motor time to process if it needs it
```
Comment (CAP1Sup) [inline src/user/main.cpp:213]: Why a little delay? The motor updates are run on interrupts, meaning that this loop doesn't run while the motor is updating.

File: src/user/main.cpp [hunk 2]
```diff
@@ -191,7 +211,12 @@ void setup() {
...
+            #ifdef ENABLE_BLINK
+            // ! Only for testing
+            blink();
+            #else
             delay(50);
+            #endif
         }
     }
```
Comment (CAP1Sup) [inline src/user/main.cpp:221]: Why a little delay? The motor updates are done in interrupts, so the motor shouldn't have any problems.

Comment (IhorNehrutsa) [reply src/user/main.cpp]: I agree with you. But delay(50) presents in the master branch.

Comment (CAP1Sup) [reply src/user/main.cpp]: Where is the delay(50)?

Comment (IhorNehrutsa) [reply src/user/main.cpp]: https://github.com/CAP1Sup/Intellistep/blob/bffbdc91923994ea7d725eebc85585b0159439e9/src/user/main.cpp#L194-L195

File: src/user/cube.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#ifndef __CUBE_H
+#define __CUBE_H
+
+void MCO_GPIO_Init(void);
+void SystemClock_Config_HSE_16M_SYSCLK_72M(void);
+void SystemClock_Config_HSE_8M_SYSCLK_72M(void);
+void SystemClock_Config_HSE_8M_SYSCLK_128M(void);
```
Comment (IhorNehrutsa) [inline src/user/cube.h:7]: I don't see 128MHz.
**HCLK 72MHz max**

https://github.com/CAP1Sup/Intellistep/pull/34

**Comment by IhorNehrutsa** [comment]
GOOD NEWS!
Latest improvements in dev get ~30 kHz input STEP pulses!
WIP

**Comment by CAP1Sup** [comment]
Awesome! Do you think that it was due to the changes with pin setting? I'm going to change the pwm pins to much lower level soon, so that should gain a bit of performance as well

**Comment by IhorNehrutsa** [comment]
Yes, the Arduino covering functions are too slow in that case.

**Comment by CAP1Sup** [comment]
I did try to get the Arduino type functions lower, but I can still take it lower than that I think. I created the analogInfo class and fasterAnalogWrite files to help with the speed. Maybe they sped it up, but I'm still going to move to basic timer manipulation soon.

**Comment by IhorNehrutsa** [comment]
It might be possible to merge this pull request and start new "Stepping Speed Improvements 2". 
This one is too big now. 
It is not completely ready, but I think this will be more comfortable way.
Thanks.

**Comment by CAP1Sup** [comment]
@IhorNehrutsa I'll merge the PR tonight. Unfortunately, there's some conflicts, so I just can't merge it now.

**Pull request merged**